App/Containers/Cafe.tsx

```diff
@@ -102,8 +135,16 @@ class Cafe extends Component<Props> {
   render() {
     // Hardcoded placeholder for services UI
     const services = ['Backup', 'Inboxing']
-    const { peer } = this.props.navigation.getParam('cafeSession').cafe
-    const buttonDisabled = !this.props.error && this.props.processing
+    const cafeSession = this.props.navigation.getParam('cafeSession')
+    const { cafe, exp } = cafeSession
+    const { peer } = cafe
+    const seconds = (exp.seconds as number) * 1000 + exp.nanos / 1000000
```

Comment:
There is a util function `Textile.util.timestampToDate` that will make this much cleaner.

Reply:
That’s convenient, thanks for the tip!